packages/server/src/server.js

```diff
@@ -114,6 +114,11 @@ export default class Server {
         if (!this.devMiddleware) {
           return next()
         }
+        // Safari over-caches JS (breaking HMR) and the seemingly only way to turn
+        // this off in dev mode is to set Vary: * header
+        if (req.url.startsWith(this.publicPath) && req.url.endsWith('.js')) {
+          res.setHeader('Vary', '*')
```

Comment:
Would cache-control: no-store be better?

Reply:
You would think so, but it's a known issue with Safari's cache mechanism (= a bug, imo). See https://stackoverflow.com/a/54337073/9614848

Reply:
@danielroe Can you please also confirm it only happens for js? (ie changing an imported asset like image accepts hmr without cache issue)

Reply:
@pi0 I've confirmed there is no issue with HMR image assets.

The issue isn't to do with all .js, just the runtime chunk. I believe the issue could be resolved by simply adding `Vary` header to `/_nuxt/runtime.js`, as that file is what determines the HMR update URLs. But I've made it more general in this PR in case of `runtimeChunk: 'multiple'`, for example.